altapay.php

```diff
@@ -0,0 +1,2353 @@
+     */
+    public function getItemCaptureRefundQuantityCount($orderLines)
+    {
+        //get the array of the itemIDs to be captured or refund of each orderline
+        $itemIDs = array_column($orderLines, 'itemId');
+        // get the array of quantities of the Items to be captured or refund of each orderline
+        $quantities = array_column($orderLines, 'quantity');
+
+        return array_combine($itemIDs, $quantities);
+    }
+    /* ******************************** */
+
+    /* Handle merchant details form */
+
+    /**
+     * Method for creating dummy order lines array in case no order lines selected for refund action
+     * @param $totalAmount
+     * @return array
+     */
+    private function createDummyOrderLinesArr($totalAmount)
+    {
+        $dummyItemOrderLine = array();
+        // Mandatory keys for orderLines:
+        $dummyItemOrderLine['description'] = 'Good-will refund';
+        $dummyItemOrderLine['itemId'] = '100200';
+        $dummyItemOrderLine['quantity'] = 1;
+        $dummyItemOrderLine['unitPrice'] = $totalAmount;
+        // Optional keys for orderLines:
+        $dummyItemOrderLine['taxAmount'] = '0.00';
+        $dummyItemOrderLine['taxPercent'] = '0.00';
+        $dummyItemOrderLine['goodsType'] = 'refund';
+
+        return array($dummyItemOrderLine);
+    }
+    /* ******************************** */
+
+    /**
+     * Handle submission of terminal form
+     */
+    private function postProcessTerminal()
+    {
+        $terminalRemoteName = $_POST['remote_name'];
+        $terminalId = getTerminalId($terminalRemoteName)[0]['id_terminal'];
+        // Update existing
+        if ($idTerminal = Tools::getValue('id_terminal')) {
+            $terminal = new Terminal((int)$idTerminal);
+        } // New
+        elseif (!($idTerminal = Tools::getValue('id_terminal')) && $terminalId) {
+            $idTerminal = $terminalId;
+            $terminal = new Terminal((int)$idTerminal);
+        } else {
+            $terminal = new Terminal;
+        }
+
+        $altapayTerminal = new AltapayTerminal();
+
+        // Currency supported?
+        if (!$altapayTerminal->hasCurrency(Tools::getValue('currency'))) {
+            $getVal = Tools::getValue('currency');
+            $this->Mhtml .=
+                sprintf('<div class="alert alert-danger">Selected terminal does not support currency %s</div>', $getVal);
+            return false;
+        }
+
+
+        // Fields
+        $fields = array('display_name', 'remote_name', 'icon_filename', 'currency', 'ccTokenControl_', 'payment_type', 'active');
+        foreach ($fields as $fieldName) {
+            $terminal->{$fieldName} = Tools::getValue($fieldName);
+        }
+
+        // Validate
+        $result = $terminal->validateFields(false, true);
+
+        if ($result == 1) {
+
+            $terminal->save();
+            return true;
+        } else {
+            $this->Mhtml .= '<div class="alert alert-danger">' . $result . '</div>';
+        }
```

Comment:
no need for else as the previous clause returns
```suggestion
        }
        
        $this->Mhtml .= '<div class="alert alert-danger">' . $result . '</div>';
```